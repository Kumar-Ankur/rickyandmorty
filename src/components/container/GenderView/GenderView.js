import React, { Component } from "react";
import { connect } from "react-redux";
import CheckBox from "../../presentation/CheckBox/CheckBox";
import * as actionCreator from "../../../actions";
import * as constants from "../../../constants";
import { getCheckBoxData } from "../../../utils";
import "./GenderView.scss";

class GenderView extends Component {
  render() {
    const { all_characters } = this.props;
    const genderData = getCheckBoxData(all_characters, constants.GENDER);
    const { addGenderFilteredData, removeGenderFilteredData } = this.props;

    const getGenderBox = genderData.map((data, index) => {
      return (
        <div key={index} data-test='container-checkbox'>
          <CheckBox
            data={data}
            addFilter={addGenderFilteredData}
            removeFilter={removeGenderFilteredData}
          />
        </div>
      );
    });

    return (
      <div className="species-content" data-test='component-gender'>
        <h2>{ constants.GENDER_FILTER}</h2>
        <form data-test='component-gender-form'>{getGenderBox}</form>
      </div>
    );
  }
}

const mapStateToProps = (state) => {
  return {
    all_characters: state.character.all_characters,
  };
};

const mapDispatchToProps = (dispatch) => {
  return {
    addGenderFilteredData: (data) =>
      dispatch(actionCreator.addGenderFilteredData(data)),
    removeGenderFilteredData: (data) =>
      dispatch(actionCreator.removeGenderFilteredData(data)),
  };
};

export default connect(mapStateToProps, mapDispatchToProps)(GenderView);
