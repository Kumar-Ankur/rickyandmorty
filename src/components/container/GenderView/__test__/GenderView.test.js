import React from "react";
import { shallow } from "enzyme";
import { applyMiddleware, createStore } from "redux";
import createSagaMiddleware from "redux-saga";
import rootReducer from "../../../../reducers";
import GenderView from "../GenderView";

const middleware = createSagaMiddleware();
const mockStore = createStore(rootReducer, applyMiddleware(middleware));
/**
 * Factory function to create a shallowwrapper for the App component.
 * @function setUp
 * @param { Object } props - Component props specific to this setup
 * @param { Object } state - Initial state for setUp
 * @returns { shallowWrapper }
 *
 */

function setUp(props = {}) {
  const wrapper = shallow(<GenderView store={mockStore} {...props} />);
  return wrapper;
}

/**
 * Return shallowWarpper containing node(s) with the given data-test attribute
 * @function findByTestAttr
 * @param {ShallowWrapper} wrapper - Enzyme shallow wrapper to search within
 * @param {string} val - value of data-test attribute for search
 * @returns { shallowWrapper}
 */
function findByTestAttr(wrapper, val) {
  return wrapper.find(`[data-test="${val}"]`);
}

describe("check GenderView component", () => {
  let wrapper;
  beforeEach(() => {
    wrapper = setUp().dive().dive();
  });

  test("generate snapshot for genderView component", () => {
    expect(wrapper).toMatchSnapshot();
  });

  test("check genderview component render without crash", () => {
    const genderComponent = findByTestAttr(wrapper, "component-gender");
    expect(genderComponent.length).toBe(1);
  });

  test("check `gender checkbox form` render without crash", () => {
    const genderFormComponent = findByTestAttr(
      wrapper,
      "component-gender-form"
    );
    expect(genderFormComponent.length).toBe(1);
  });

  test("check `getCheckBoxData` call without error", () => {
    const getCheckBoxDataMock = jest.fn();
    const instance = wrapper.instance();
    instance.getCheckBoxData = getCheckBoxDataMock;
    instance.getCheckBoxData();
    expect(getCheckBoxDataMock).toHaveBeenCalled();
  });

  test("check `addGenderFilteredData` call without error", () => {
    const addGenderFilteredDataMock = jest.fn();
    const instance = wrapper.instance();
    instance.addGenderFilteredData = addGenderFilteredDataMock;
    instance.addGenderFilteredData();
    expect(addGenderFilteredDataMock).toHaveBeenCalled();
  });

  test("check `removeGenderFilteredData` call without error", () => {
    const removeGenderFilteredDataMock = jest.fn();
    const instance = wrapper.instance();
    instance.removeGenderFilteredData = removeGenderFilteredDataMock;
    instance.removeGenderFilteredData();
    expect(removeGenderFilteredDataMock).toHaveBeenCalled();
  });

  test("check `getGenderBox` call without error", () => {
    const getGenderBoxMock = jest.fn();
    const instance = wrapper.instance();
    instance.getGenderBox = getGenderBoxMock;
    instance.getGenderBox();
    expect(getGenderBoxMock).toHaveBeenCalled();
  });
});
