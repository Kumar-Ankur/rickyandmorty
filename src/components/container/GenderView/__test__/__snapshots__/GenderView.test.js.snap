// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`check GenderView component generate snapshot for genderView component 1`] = `ShallowWrapper {}`;
